### PR TITLE
Implement AV1 codec support

### PR DESCRIPTION
**Pull request by CastagnaIT**
This implement AV1 codec support

What is the current status, has been implemented for DASH and unofficially for HSL
the unencrypted streams are played with success, some examples:

https://storage.googleapis.com/bitmovin-demos/av1/stream.mpd
https://storage.googleapis.com/bitmovin-demos/av1/stream_chrome.mpd
http://ftp.itec.aau.at/datasets/mmsys18/testing/Beauty_4sec/multi-codec.mpd

For encrypted AV1 streams i havent found sample streams to test,
the only way that i found to test it is by using n@tflix, that provide MP4/CBCS.
Unfortunately, these encrypted streams are not being played in the right way,
~a possible cause is the use of CMAF~ (EDIT: the problem is wrong video format set), the image result as:
immagine
this problem looks similar to my old test with VP9.2 in Issue https://github.com/xbmc/inputstream.adaptive/issues/525 (still unresolved, not more tested over the time not knowing how to solve)

To do tests with N@tflix is possible to use this PR branch:
https://github.com/CastagnaIT/plugin.video.netflix/pull/1343
(clearly on browser Chrome N@tflix AV1 playback works fine)

I can suggest to add this support to allow to play at least AV1 unencrypted streams,
and when we will find an encrypted sample in future we will see how goes

**This depends from:** https://github.com/xbmc/xbmc/pull/21390

### CODE REVIEW

**Comment by glennguy** [comment]
Is there any info here that could help - https://aomediacodec.github.io/av1-isobmff/

~~One thing I've noticed is that although we seem to need to user codec private data, this info looks like it would stay empty in the sample streams. Maybe we need to extract it from the init segment if it's not in the manifest?~~ nevermind, re-read the OP and can see it works for unencrypted sample streams.

**Comment by CastagnaIT** [comment]
already tried read in link but i have found nothing that help me
i am thinking of something not implemented in wvdecrypter video decode but i dont know what
(this codec do not have things like sps or pps as hevc is much simplier)
another think, that i dont know if it is relevant is that n@tflix have AV1 with CMAF
we should test an encrypted AV1 without CMAF and see how works

**Comment by AlwinEsch** [comment]
@CastagnaIT please make a rebase and to combine with https://github.com/xbmc/inputstream.adaptive/pull/989 to have ready about new release.

**NOTE:** We need to wait until launchpad have newest Kodi Version before we can merge requests on add-ons!

**Comment by AlwinEsch** [comment]
@glennguy has brought in the other, seen are a bit different, @CastagnaIT to become Jenkins happy you need to make again, Sorry!

**Comment by CastagnaIT** [comment]
yes now i will see to fix it

**Review by glennguy — APPROVED** [review]
Thanks @CastagnaIT 
The wvdecrypter changes in an upcoming PR should address the CBCS/multi-subsample issues

**Pull request merged**
